config/v1/types_infrastructure.go

```diff
@@ -144,6 +148,15 @@ type GCPPlatformStatus struct {
 	Region string `json:"region"`
 }
 
+// OpenStackPlatformStatus holds the current status of the OpenStack infrastructure provider.
+type OpenStackPlatformStatus struct {
+	// APIVIP is the VRRP address for the internal OpenShift API access.
+	APIVIP string `json:"apiVip,omitempty`
+
+	// DNSVIP is the VRRP address for the internal cluster DNS.
+	DNSVIP string `json:"dnsVip,omitempty`
```

Comment:
See https://github.com/openshift/api/pull/348#discussion_r301569861

I have the same question here about what DNS service this IP refers to, who provides and consumes the value, etc.

Reply:
@ironcladlou the answers are pretty much the same (we're facing similar issues and solving them in a similar manner as baremetal and ovirt).

Unlike AWS, none of these platforms can expect a load balancer or DNS-as-a-service and so they have to provide the underlying DNS (e.g. the `api` & `api-int` endpoints and the SRV etcd records) some other way. The solution we've all converged on is that each node runs (and relies) on its own coredns static pod with mdns providing dynamic updates and keepalived for IP failover.

You make good points about differentiating between this and the DNS service running inside the cluster as well as the VIP/VRRP implementation details leaking in.

Would something like this work?

```
// ApiIntIp is an IP address managed by the OpenStack provider backing the internal `api-int` record
ApiIntIp string `json:"apiIntIp,omitempty`

// ProviderDnsIp is the IP address for the internal DNS used by the nodes. Unlike the one managed by the DNS operator, `ProviderDnsIp` provides name resolution for the nodes themselves.
ProviderDnsIp string `json:"providerDnsIp,omitempty`
```

(I agree with the importance of choosing the right names and descriptions, but I'd rather not spend too long on this as the 4.2 feature freeze is looming over us)

Reply:
Oh, and the values are consumed by the MCO, which provides the configuration for the static pods we need to run (and it's those extra static pods that need these IPs).

Reply:
@tomassedovic 

+1, understood, and I think your suggestion does a good job disambiguating. I just want to make sure I understood what these things mean and whether they implied some missed requirements in other components which might need to consume them.

>(I agree with the importance of choosing the right names and descriptions, but I'd rather not spend too long on this as the 4.2 feature freeze is looming over us)

Sure, I don't want to get in the weeds rehashing the same points here. I personally don't want to block either PR on the subjective naming stuff.

Reply:
@ironcladlou thanks, that makes perfect sense.

I've updated the PR, please let me know if there's anything else you need.